feat: add C implementation for `stats/base/dists/levy/quantile` (#4756)
stdlib-js/stdlib: lib/node_modules/@stdlib/stats/base/dists/levy/quantile/README.md

```

#### stdlib_base_dists_levy_quantile( p, mu, c )

Evaluates the [quantile function][quantile-function] for a [Lévy][levy-distribution] distribution with parameters `mu` (location parameter) and `c` (scale parameter).

```c
double out = stdlib_base_dists_levy_quantile( 0.8, 0.0, 1.0 );
// returns ~15.58
```

The function accepts the following arguments:

-   **p**: `[in] double` probability.
-   **mu**: `[in] double` location parameter.
-   **c**: `[in] double` scale parameter.

```c
double stdlib_base_dists_levy_quantile( const double p, const double mu, const double c );
```

</section>

<!-- /.usage -->

<!-- C API usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="notes">

</section>

<!-- /.notes -->

<!-- C API usage examples. -->

<section class="examples">

### Examples

```c
#include "stdlib/stats/base/dists/levy/quantile.h"
#include <stdlib.h>
#include <stdio.h>

static double random_uniform( const double min, const double max ) {
    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
    return min + ( v*(max-min) );
}

int main( void ) {

>     double p;
>     double mu;

Reviewer: ```suggestion
    double mu;
    double p;
```

<!-- /.examples -->

<!-- C interface documentation. -->

* * *

<section class="c">

## C APIs

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- C usage documentation. -->

<section class="usage">

### Usage

```c
#include "stdlib/stats/base/dists/levy/quantile.h"
```

#### stdlib_base_dists_levy_quantile( p, mu, c )

> Returns the quantile function for a Lévy distribution with location parameter `mu`  and scale parameter `c`.

Reviewer 2: ```suggestion
Evaluates the [quantile function][quantile-function] for a [Lévy][levy-distribution] distribution with parameters `mu` (location parameter) and `c` (scale parameter).
```

> Returns the [quantile function][quantile-function] for a [Lévy][levy-distribution] distribution with parameters `mu` (location parameter) and `c` (scale parameter) at a probability `p`.

Reviewer 2: ```suggestion
Returns the quantile function for a Lévy distribution with location parameter `mu`  and scale parameter `c`.
```
should match the JSDoc API call mentioned above.
SInce you might have referenced it in `main.c` and `quantile.h` you'll have to change it there as well.

Author: thanks for review . I am doing any more changes if needed then tell me.

```c

> double out = stdlib_base_dists_levy_quantile( 1.1, 0.0, 1.0 );
> // returns NaN

Reviewer 2: ```suggestion
double out = stdlib_base_dists_levy_quantile( 0.8, 0.0, 1.0 );
// returns ~15.58
```
In my opinion instead of adding an example which returns `NaN` we can switch to another example which can display an approx value. By doing this we can have a better idea of the function.

```

The function accepts the following arguments:

-   **p**: `[in] double` probability.
-   **mu**: `[in] double` location parameter.
-   **c**: `[in] double` scale parameter.

```c
double stdlib_base_dists_levy_quantile( const double p, const double mu, const double c );
```

</section>

<!-- /.usage -->

<!-- C API usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="notes">

</section>

<!-- /.notes -->

<!-- C API usage examples. -->

<section class="examples">

### Examples

```c

> #include "stdlib/stats/base/dists/levy/quantile.h"
> 
> #include <stdlib.h>
> #include <stdio.h>

Reviewer 2: ```suggestion
#include "stdlib/stats/base/dists/levy/quantile.h"
#include <stdlib.h>
#include <stdio.h>
```

static double random_uniform( const double min, const double max ) {
    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
    return min + ( v*(max-min) );
}

int main( void ) {
    double p;
    double mu;
    double c;
    double y;
    int i;
    for ( i = 0; i < 25; i++ ) {
        p = random_uniform();
        mu = random_uniform( 0.0, 10.0 ) - 5.0;

>         c = random_uniform( 0.0, 20.0 );

Reviewer 2: add the updated `example.c` file in the markdown file,

Author: done

>         y = stdlib_base_dists_levy_quantile( mu, c );

Reviewer 2: ```suggestion
        y = stdlib_base_dists_levy_quantile( p, mu, c );
```
